code-challenges.test.js

```diff
@@ -10,50 +10,159 @@
+// within the .map function, we will use concatenation to return the character at the zero index of the string, uppercased via the .toUpperCase method, then the rest of the string, with the first character removed via the .slice method.
+
+const capitalizer = (array) =>
+  array.map((string) => string[0].toUpperCase() + string.slice(1));
+// Test Suites: 1 passed, 1 total
+// Tests:       2 passed, 2 total
 
 // --------------------3) Create a function that takes in a string and logs the index of the first vowel.
 
 // a) Create a test with expect statements for each of the variables provided.
-
-const vowelTester1 = "learn"
+describe("firstVowel", () => {
+  it("takes in a string and logs the index of the first vowel.", () => {
+    expect(firstVowel(vowelTester1)).toEqual(1);
+    expect(firstVowel(vowelTester2)).toEqual(0);
+    expect(firstVowel(vowelTester3)).toEqual(2);
+  });
+});
+
+const vowelTester1 = "learn";
 // Expected output: 1
-const vowelTester2 = "academy"
+const vowelTester2 = "academy";
 // Expected output: 0
-const vowelTester3 = "challenges"
+const vowelTester3 = "challenges";
 // Expected output: 2
 
+// ReferenceError: firstVowel is not defined
+
 // b) Create the function that makes the test pass.
+// Declare a function name firstVowel that has a parameter of a string.
+// Use a for loop to iterate through the string, starting at an index of 0, stopping at the index 1 less than the length of the string, and increasing by 1 each time.
+// At each iteration, run an if statement:
+//If the value at the current index is equal to any vowel, return the current index
+
+/*
+const firstVowel = (string) => {
+    let placeholder = "no vowels"
+    for (let i=0; i< string.length; i++){
+        if(string[i] === "a" || "e" || "i" || "o" || "u"){
+            placeholder = i
+            console.log(i)
+        } 
+        return placeholder
+    }
+}
+ This is not working for me! I keep getting this error:
+   firstVowel
+    ✕ takes in a string and logs the index of the first vowel. (51 ms)
+
+  ● firstVowel › takes in a string and logs the index of the first vowel.
+
+    expect(received).toEqual(expected) // deep equality
+
+    Expected: 1
+    Received: 0
+
+    In my mind, the logic should only replace the placeholder with 0 if the letter at index 0 is one of the strings listed in the if statement. But it seems to be updating the value of placeholder even though it is a consonant! I'm going to try another way.
+
+ New idea! this time the function will create variables for the index of the first occurence of every vowel. Then I will place those indexes into an array. I will filter the array to remove -1, since that indicates the vowel was not present. from the filtered array, I will sort it from smallest to largest, and from that array, get the value at the index of zero, which will be the smallest number and hence the first vowel from the string.
```

Comment:
Nice work in displaying your first process, and rewriting pseudocode for the second approach.   You asked in your comments for the PR about other ways to approach this such as reg ex.  Reg ex combined with search is a great approach, however, I appreciate that you held off on using it before you are comfortable with it.   Regular expressions allow you to be able to check against a collection of characters.  Here is a great article to help break it down: [Reg Ex](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions)